drivers/timer/nrf_rtc_timer.c

```diff
@@ -20,8 +20,6 @@
 		      / CONFIG_SYS_CLOCK_TICKS_PER_SEC)
 #define MAX_TICKS ((COUNTER_MAX - CYC_PER_TICK) / CYC_PER_TICK)
 
-#define MIN_DELAY 32
-
 static struct k_spinlock lock;
 
 static u32_t last_count;
```

Comment:
If it is possible that RTC CC is reconfigured before then there is a chance that previous alarm will generate interrupt. NOt sure if it is a problem here (it may mean only once additional interrupt without consequences). If so then we don't need to trouble else here is the algorithm i came up with to handle that (it is not that simple since event may be generated after changing CC (if it happens up to 15us before previous CC deadline).

https://github.com/zephyrproject-rtos/zephyr/blob/6bdbf47b2acf69533e26bb500b3a75f4ffb25085/drivers/counter/counter_nrfx_rtc.c#L188